koku/masu/database/koku_database_access.py

```diff
@@ -113,3 +115,54 @@ def delete(self, obj=None):
             deleteme = self._obj
         with schema_context(self.schema):
             deleteme.delete()
+
+
+def mtd_check_remainder(base_select_query):
+    return base_select_query.count()
+
+
+def mini_transaction_delete(base_select_query):
+    """
+    Uses a select query to make a mini transactinal delete loop.
+    Schema should be set before calling this function.
+    Args:
+        base_select_query (QuerySet) : Single-table django select query
+    Returns:
+        tuple : (deleted_record_total, records_remaining)
+    """
+    del_record_limit = int(os.getenv("DELETE_CYCLE_RECORD_LIMIT", 5000))
```

Comment:
To be extra Django-y, we could move these to settings.py and `from django.conf import settings` and reference them from settings like `settings. DELETE_CYCLE_RECORD_LIMIT` but I suspect we'll be changing this again soon to only remove expired partitions, so optional.